File: src/test/java/com/josephbaca/world/CoordinateGridTest.java
package com.josephbaca.world;

import static org.junit.jupiter.api.Assertions.*;

import com.josephbaca.util.ContextManager;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

class CoordinateGridTest {

  private ContextManager contextManager;
  private World world;

  @BeforeEach
  void setUp() {
    contextManager = new ContextManager();
    world = new World("testworld", 10, 10, contextManager);
  }

  @Test
  void setCoordinate() {
    CoordinateGrid<Room> grid = new CoordinateGrid<>(2, 2);

    for (int i = 0; i < grid.sizeX(); i++) {
      for (int j = 0; j < grid.sizeX(); j++) {
        grid.setCoordinate(Coordinate.of(i, j), new Room(contextManager, 5, 5, world));
      }
    }

    // Sanity check
    assertEquals("R R\nR R", grid.toDisplayString());

    // Set top left
    grid.setCoordinate(Coordinate.of(0, 1), new Room(contextManager, 5, 5, world, "T"));
    assertEquals("T R\nR R", grid.toDisplayString());
  }

  @Test
  void toDisplayString() {
    CoordinateGrid<Room> grid = new CoordinateGrid<>(2, 2);

    for (int i = 0; i < grid.sizeX(); i++) {
      for (int j = 0; j < grid.sizeX(); j++) {
        grid.setCoordinate(Coordinate.of(i, j), new Room(contextManager, 5, 5, world));
      }
    }

    assertEquals("R R\nR R", grid.toDisplayString());
  }
}

File: src/test/java/com/josephbaca/world/CoordinateGridTest.kt
package com.josephbaca.world

import com.josephbaca.context.ContextManager
import com.josephbaca.context.Room
import com.josephbaca.context.World
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

internal class CoordinateGridTest {

    private var contextManager: ContextManager? = null
    private var world: World? = null

    @BeforeEach
    fun setUp() {
        contextManager = ContextManager()
        world = World(10, 10, contextManager!!)
    }

    @Test
    fun setCoordinate() {
        val grid = CoordinateGrid<Room>(2, 2)

        for (i in 0 until grid.sizeX) {
            for (j in 0 until grid.sizeX) {
                grid.setCoordinate(Coordinate(i, j), Room(contextManager!!, world!!))
            }
        }

        // Sanity check
        assertEquals("R R\nR R", grid.toDisplayString())

        // Set top left
        grid.setCoordinate(Coordinate(0, 1), Room(contextManager!!, world!!, "T"))
        assertEquals("T R\nR R", grid.toDisplayString())
    }

    @Test
    fun toDisplayString() {
        val grid = CoordinateGrid<Room>(3, 2)
        grid.sizeX

        for (i in 0 until grid.sizeX) {
            for (j in 0 until grid.sizeY) {
                grid.setCoordinate(Coordinate(i, j), Room(contextManager!!, world!!))
            }
        }

        assertEquals("R R R\nR R R", grid.toDisplayString())
    }
}